java/maven.gradle

```diff
@@ -6,6 +6,10 @@ def groupId = "io.ably"
 def artifactId = "ably-java"
 def localReleaseDest = "${buildDir}/release/${version}"
 
+task promoteRepo {
+	println("HELLO PROMOTE REPO")
+}
+
```

Comment:
@paddybyers I'm trying to write those ``releaseToMaven`` tasks. You see, I have the same task, ``promoteRepo`` declared for both ``andoird`` and ``java`` subprojects.

I'm running ``gradle :java:promoteRepo`` and gradle also runs the android task, when I would expect only the ``java`` one to run. Running ``gradle :android:promoteRepo`` behaves the same.

On the other hand, playing with the existing ``uploadArchives`` task behaves as expected:

* ``gradle uploadArchives`` runs both subprojects and both ``ably-java`` and ``ably-android`` are uploaded to nexus.

* ``gradle :java:uploadArchives`` only uploads ``ably-java``
 
* ``gradle :android:uploadArchives`` only uploads ``ably-android``.

Do you happen to know what am I missing? Or someone from the team who knows gradle better?

Reply:
> I'm running `gradle :java:promoteRepo` and gradle also runs the android task, when I would expect only the `java` one to run. Running `gradle :android:promoteRepo` behaves the same.

I think that's because what you wrote was the task definition, which will run as part of the gradle build always. That's not the same as executing the task. That seems to be possible as a relatively simple sequence of subtasks via eg https://stackoverflow.com/a/32909428. However, lets focus on getting the existing release task(s) working first.